In [681]:
import sys
import os
import pandas as pd
import numpy as np
import datetime as dt
import matplotlib.pyplot as plt
import h5py
import json
import requests
import base64
sys.path.append('../')

pd.options.display.max_rows=99

In [682]:
github_userName = 'Tanag3r'
github_token = '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'
ebird_token = 'j6c7l80ga2ib'

In [683]:
##pull in trailhead stops
trailheadRef = pd.DataFrame()
trailheadRef_json = 'https://raw.githubusercontent.com/Tanag3r/trailheadDirect_birds/cooking_amerRob_analysis/trailheadsRef.json'
response = requests.get(trailheadRef_json)
trailheadRef = pd.DataFrame(response.json())
trailheadRef.set_index('StopName')

##check length of lat and long, each must be at least 4 char

Route  \
StopName                                 
EastSunsetWay             IssaquahAlps   
HighSchool                IssaquahAlps   
ChiricoTrail_PooPooPoint  IssaquahAlps   
SquakMountain             IssaquahAlps   
MargaretsWay              IssaquahAlps   
MountSi                        MountSi   
MountTeneriffe                 MountSi   
LittleSi                       MountSi   

                                                                    Address  \
StopName                                                                      
EastSunsetWay                      661-831 E Sunset Way, Issaquah, WA 98027   
HighSchool                Parking lot, The Rainier Trail, Issaquah, WA 9...   
ChiricoTrail_PooPooPoint  11400 Issaquah-Hobart Road Southeast, Issaquah...   
SquakMountain                13201 Squak Mountain Rd SE, Issaquah, WA 98027   
MargaretsWay                               190th Ave SE, Issaquah, WA 98027   
MountSi                                               King County, WA 98104   
MountTeneriffe                     Mount Teneriffe Rd, North Bend, WA 98045   
LittleSi                                  SE Mt Si Rd, North Bend, WA 98045   

                           Latitude   Longitude  
StopName                                         
EastSunsetWay             47.529635 -122.025119  
HighSchool                47.519345 -122.029801  
ChiricoTrail_PooPooPoint  47.499949 -122.021730  
SquakMountain             47.481465 -122.053997  
MargaretsWay              47.506620 -122.086660  
MountSi                   47.488966 -122.723044  
MountTeneriffe            47.490104 -122.709182  
LittleSi                  47.489366 -122.753833

In [684]:
##fetch all hotspots within 4 kilometers of each trailhead

trailheadHotspots = []

for StopName in trailheadRef.itertuples():
    trailhead_stopName = str(StopName.StopName)
    url = '''https://api.ebird.org/v2/ref/hotspot/geo?lat={}&lng={}&dist=4&fmt=json'''.format(StopName.Latitude,StopName.Longitude)
    ebirdapi_auth_header = {'X-eBirdApiToken': ebird_token}
    ebird_request = requests.get(url,headers=ebirdapi_auth_header)
    ebird_response = pd.DataFrame(ebird_request.json()).insert(0,'stopName',pd.Series(trailhead_stopName))
    trailheadHotspots.append(ebird_response)
    ##trailheadHotspots.extend(ebird_response,trailhead_stopName)

##after checking length of lat and long, pass concat values into the nearby hotspot api

base_url = 'https://api.ebird.org/v2/ref/hotspot/geo?'
ebirdapi_auth_header = {'X-eBirdApiToken': ebird_token}
ebirdapi_params = {
    'lat': str(47.529635),
    'lng': str(-122.025119),
    'dist': str(4),
    'back': str(30),
    'fmt': 'json'
}

response = requests.get(base_url,params=ebirdapi_params,headers=ebirdapi_auth_header)
print(response.json())

##testing adding stopname to df

jointests = []

for StopName in trailheadRef.itertuples():
    url = 'https://api.ebird.org/v2/ref/hotspot/geo?lat={}&lng={}&dist=4&format=json'.format(StopName.Latitude,StopName.Longitude)
    stopName = StopName.StopName
    jointest = pd.DataFrame(data=[(url,stopName)],columns=['url','StopName'])
    jointests.append(jointest)

print(jointests)

In [685]:
all_trailheadHotspots = pd.concat(trailheadHotspots)

all_trailheadHotspots.head(20)

ValueError: All objects passed were None

In [ ]:
all_trailheadHotspots.shape

(63, 72)

all_trailheadHotspots.to_csv('trailheadHotspots_2022JAN.csv',sep=',',index=False)